server/src/main/java/org/opensearch/common/time/OpenSearchDateTimeFormatter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.time;
+
+import java.text.Format;
+import java.text.ParsePosition;
+import java.time.ZoneId;
+import java.time.chrono.Chronology;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.time.temporal.TemporalQuery;
+import java.util.Locale;
+
+/**
+* Wrapper class for DateTimeFormatter{@link java.time.format.DateTimeFormatter}
+* to allow for custom implementations for datetime parsing/formatting
+ */
+public class OpenSearchDateTimeFormatter implements OpenSearchDateTimePrinter {
```

Comment:
It doesn't need to be public, right?
```suggestion
class OpenSearchDateTimeFormatter implements OpenSearchDateTimePrinter {
```